lib/model-datastore.js

```diff
@@ -82,75 +93,118 @@ function toDatastore(obj, nonIndexed) {
   return results;
 }
 
-// Lists all Entities in the Datastore sorted alphabetically by title.
-// The ``limit`` argument determines the maximum amount of results to
-// return per page. The ``token`` argument allows requesting additional
-// pages. The callback is invoked with ``(err, Entities, nextPageToken)``.
-function list(kind, limit, token, cb) {
-  const q = ds.createQuery([kind])
-    .limit(limit)
-    .start(token);
+/**
+ * Lists all Entities in the Datastore sorted alphabetically by title.
+ * The ``limit`` argument determines the maximum amount of results to
+ * return per page. The ``token`` argument allows requesting additional
+ * pages. The callback is invoked with ``(err, Entities, nextPageToken)``.
+ *
+ * @param {string} kind
+ * @param {number} limit
+ * @param {number} token
+ * @return {Promise<Array<Object>>}
+ */
+function list(kind, limit, token) {
+  return new Promise((resolve, reject) => {
+    const q = ds.createQuery([kind])
+      .limit(limit)
+      .start(token);
 
-  ds.runQuery(q, (err, entities, nextQuery) => {
-    if (err) {
-      return cb(err);
-    }
-    const hasMore = entities.length === limit ? nextQuery.startVal : false;
-    cb(null, entities.map(fromDatastore), hasMore);
+    ds.runQuery(q, (err, entities, nextQuery) => {
+      if (err) {
+        return reject(err);
+      }
+      const hasMore = entities.length === limit ? nextQuery.startVal : false;
+      resolve(entities.map(fromDatastore), hasMore);
```

Comment:
You are right. Should probably `resolve({entities: entities.map(fromDatastore), hasMore:hasMore};`